mobile_packet_verifier/src/event_ids.rs

```diff
@@ -0,0 +1,73 @@
+use chrono::{DateTime, Duration, Utc};
+use sqlx::{Pool, Postgres, Transaction};
+
+use crate::settings::Settings;
+
+pub async fn is_duplicate(
+    tx: &mut Transaction<'_, Postgres>,
+    event_id: String,
+) -> anyhow::Result<bool> {
+    let result =
+        sqlx::query_scalar::<_, bool>("SELECT EXISTS(SELECT 1 FROM event_ids WHERE event_id = $1)")
+            .bind(event_id.clone())
+            .fetch_one(&mut *tx)
+            .await?;
+
+    Ok(result)
+}
+
+pub async fn record(
+    tx: &mut Transaction<'_, Postgres>,
+    event_id: String,
+    received_timestamp: DateTime<Utc>,
+) -> anyhow::Result<()> {
+    sqlx::query("INSERT INTO event_ids(event_id, received_timestamp) VALUES($1,$2)")
+        .bind(event_id)
+        .bind(received_timestamp)
+        .execute(tx)
+        .await
+        .map(|_| ())
+        .map_err(anyhow::Error::from)
+}
```

Comment:
worth the speed boost to make this an upsert statement returning whether or not the insert happened and then do it in a single query?

Reply:
i've done this in a different app which attempts to do an upsert and then returns the boolean if the row was inserted or not
```
    query_with_bindings.execute(&mut tx).await?.rows_affected() > 0;
```

Reply:
I agree with @jeffgrunewald here, this should be an upsert

Reply:
This is would the query would look like:
```rust
let is_duplicate = sqlx::query(
    r#"
    INSERT INTO event_ids (event_id, received_timestamp) VALUES ($1, $2) 
    ON CONFLICT DO NOTHING
    "#
)
.bind(event_id)
.bind(received_timestamp)
.execute(tx)
.await?
.rows_affected() > 0;
```

Reply:
Agreed, upsert is better 5c66469